src/NuGet.Services.EndToEnd/EnvironmentSettings.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+
+namespace NuGet.Services.EndToEnd
+{
+    public static class EnvironmentSettings
+    {
+        private static EnvironmentVariableTarget[] Targets = new[]
+        {
+            EnvironmentVariableTarget.Process,
+            EnvironmentVariableTarget.User,
+            EnvironmentVariableTarget.Machine
+        };
+
+        public static string GalleryBaseUrl => GetEnvironmentVariable("GalleryBaseUrl", required: true);
+
+        public static string[] TrustedHttpsCertificates
+        {
+            get
+            {
+                var value = GetEnvironmentVariable("TrustedHttpsCertificates", required: false);
+                if (value == null)
```

Comment:
Also I would probably name this variable something else--naming it `value` like a setter makes this slightly confusing.